### DISCUSSION TITLE
Stun Click isn't stunning no more?

### DISCUSSION THREAD

**Discussion by Dexteriority** [winstxnhdw/lc-hax#395] [Q&A] [answered]
I just used this feature after requesting a change for Stun hax with Radar Flash instead of flashbang, but now they are not stopping at allXD.

I bought a radar booster just in case it might triggers with Radar Booster but nope.
I click M1, M2, and M3 with /stunclick enabled but no enemies getting stun xd.

Previous version with infinity flashbang mechanic, the enemies has stun animation even though when they are not stopping. But this one have no animation and not getting stun.

Also the **/stun** works but it stun **ALL** of the enemies instead of changing the duration of the stunclick like the usual.

**Comment by winstxnhdw** [comment]
Oh, I didn’t change `stunclick`. I only changed `stun` because that was what you requested previously.

**Reply by Dexteriority** [reply]
but the Stunclick doesn't work either? I used **/stunclick** it says enabled but i clicked and the enemies didn't stop or has stun animation.
If the **/stun** is a command to change how long the **/stunclick**  will be, its fine as it is :D 

But the change i was talking about is the **/stunclick** behaviour from infinite flashbang to Radar Flash, if this is going to be a hell of a code to mess with, we can revert back to original :D

**Reply by winstxnhdw** [reply, marked as answer]
`stun` and `stunclick` are totally separate commands. `stun` diffuses a stun from your camera at supposedly infinite distance. `stunclick` on the otherhand, is supposed to stun or jam an enemy or object for 5 seconds. This duration is not negotiable.

**Reply by winstxnhdw** [reply]
Also, `stunclick` requires that you equip a weapon before using it.

**Reply by Dexteriority** [reply]
NO WAY! from the start i thought **/stun** is functioning as the duration for the **/stunclick**.
I misunderstood these two from the beginning! :O

**Reply by Dexteriority** [reply]
Anyway, is the **/stun** actually meant to stun literally every enemies? or the only in my current view?
Because i used it and all enemies in the planet got stun. 

And is the **/stunclick** requirements of equip a weapon before using it was there from the start? I don't remember that how it used to be🤔🤔

**Reply by winstxnhdw** [reply]
It's meant to stun every enemy.

**Reply by winstxnhdw** [reply]
> And is the /stunclick requirements of equip a weapon before using it was there from the start? I don't remember that how it used to be🤔🤔

Yeap, for a really long time too. One of the very first few contributors, @armorfid, added it.

**Reply by Dexteriority** [reply]
Somehow i remember it like i can just stun them instantly without equiping anything earlier XD

**Reply by Dexteriority** [reply]
Hey so i went into Eclipsed moon and bought a Shovel, enabled /stunclick command, equiping shovel and dropped it, pressed the M1 towards the dogs, nothing happens?
They aren't stopping :((

**Reply by winstxnhdw** [reply]
You gotta have it equipped constantly.

**Reply by Dexteriority** [reply]
I had reserved weapon mod on I did have it in my side inventory. doesn't work, but i will try vanilla

**Reply by winstxnhdw** [reply]
It has to fulfill the condition that `currentlyHeldObjectServer` is a defensive object.

**Reply by Dexteriority** [reply]
my brain ain't functioning right now, normal scraps is defensive object right?

**Reply by winstxnhdw** [reply]
No. It just means any weapon.

**Reply by Dexteriority** [reply]
oh yeah it does work after i held a normal scraps(tea kettle), but is it possible to use it with empty hands? or it'll just confuse the script?

**Reply by Dexteriority** [reply]
huh?! Tea Kettle is considered as weapon?! XD

**Reply by Dexteriority** [reply]
Well any items that can't be use to attack are able to perform this command successfully, thank you!
